sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/workflowentity/WorkflowEntity.java

```diff
@@ -356,6 +505,67 @@ public static WorkflowEntity.StepBuilder step(String name) {
     return new WorkflowEntity.StepBuilder(name);
   }
 
+  public static class StepConfig {
+    public final String stepName;
+    public final Optional<Duration> timeout;
+    public final Optional<RecoverStrategy<?>> recoverStrategy;
+
+    public StepConfig(String stepName, Optional<Duration> timeout, Optional<RecoverStrategy<?>> recoverStrategy) {
+      this.stepName = stepName;
+      this.timeout = timeout;
+      this.recoverStrategy = recoverStrategy;
+    }
+  }
+
+  public static class RecoverStrategy<T> {
+
+    public final int maxRetries;
+    public final String failoverToStepName;
+    public final Optional<T> failoverStepInput;
+
+    public RecoverStrategy(int maxRetries, String failoverToStepName, Optional<T> failoverStepInput) {
+      this.maxRetries = maxRetries;
+      this.failoverToStepName = failoverToStepName;
+      this.failoverStepInput = failoverStepInput;
+    }
+
+    /**
+     * Retry strategy without failover configuration
+     */
+    public static class MaxRetriesRecoverStrategy {
+      public final int maxRetries;
+
+      public MaxRetriesRecoverStrategy(int maxRetries) {
+        this.maxRetries = maxRetries;
+      }
+
+      /**
+       * Once max retries is exceeded, transition to a given step name.
+       */
+      public RecoverStrategy failoverTo(String stepName) {
+        return new RecoverStrategy(maxRetries, stepName, Optional.<Void>empty());
+      }
+
+      /**
+       * Once max retries is exceeded, transition to a given step name with input parameter.
+       */
+      public <T> RecoverStrategy<T> failoverTo(String stepName, T input) {
+        return new RecoverStrategy(maxRetries, stepName, Optional.of(input));
+      }
+
+      public int getMaxRetries() {
+        return maxRetries;
+      }
+    }
+
+    /**
+     * Set the number of retires, <code>maxRetries</code> equals 0 means that the step won't retry in case of failure.
+     */
+    public static MaxRetriesRecoverStrategy maxRetries(int maxRetries){
+      return new MaxRetriesRecoverStrategy(maxRetries);
+    }
```

Comment:
What about a `failoverTo` at that level? 
It could return a `RecoverStrategy` with a `maxRetry` set to 0. WDYT?

```suggestion
    }
    
    public <T> RecoverStrategy<T> failoverTo(String stepName, T input) {
      return new RecoverStrategy(0, stepName, Optional.of(input));
    }
```